### PR TITLE
DPF-1838 - Added unit tests, addressed multiple issues. Default hash …

### PR DESCRIPTION
**Pull request by bcam909**
…processor changed to SHA256, _message removed from jsonMessageBuilderProvider, withOverride deprecated, etc.

This addresses #4 #5 and #6

### CODE REVIEW

File: src/test/java/com/godaddy/logging/logstash/LogstashTests.java [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+    public void test_cycles() throws IOException {
+        CycleObject cycleObject = new CycleObject();
+
+        cycleObject.text = "test_cycles";
+
+        cycleObject.cycle = cycleObject;
+
+        logger.with(cycleObject).info("Cycles");
+
+        assertEquals(getMarkers(), "{\n" +
```
Comment (devshorts) [inline src/test/java/com/godaddy/logging/logstash/LogstashTests.java]: Can you also test the log message is where we expect it to be?

File: src/main/java/com/godaddy/logging/LoggingConfigs.java [hunk 2]
```diff
@@ -267,6 +267,16 @@ public LoggingConfigs withExceptionTranslator(ExceptionTranslator exceptionTrans
...
+    public <T> LoggingConfigs addOverride(Class<T> clazz, Function<T, String> mapper) {
+        customMapper.put(clazz, (Function<Object, String>) mapper);
+
+        return this;
+    }
+
+    /**
+     * Deprecated, use addOverride instead.
+     */
+    @Deprecated
```
Comment (devshorts) [inline src/main/java/com/godaddy/logging/LoggingConfigs.java]: Lets add a reason for the deprecation. We want "with" to move to all immutables, and this one is mutating so it breaks our standards.

**Comment by devshorts** [comment]
LGTM. Lets get another approval from @jakeswenson before merge.

**Comment by jakeswenson** [comment]
:heavy_check_mark: LGTM

**Pull request merged**